src/components/Application.tsx

```diff
@@ -401,10 +402,15 @@ export class Application extends React.Component<Props, State> {
       jobs: this.state.jobs.$filter(j => j.id !== id),
     })
     if (this.state.route.jobIds.includes(id)) {
+      const job = this.state.jobs.records.find(j => j.id === id)
       this.navigateTo({
         pathname: this.state.route.pathname,
         search: this.state.route.search.replace(new RegExp('\\??jobId=' + id), ''),
       })
+      let forgetPromise = forgetJob(id)
+      forgetPromise.catch(() => {
+        this.handleJobCreated(job)
```

Comment:
Can we add code to explicitly add this record back to the jobs list?  Using `handleJobCreated` is a risky choice because the implementation may end up getting more complicated and new-job-specific logic that we don't want happening to a job that fails to be forgotten.